### DISCUSSION TITLE
Indicator constraints

### DISCUSSION THREAD

**Discussion by buddih09** [XiongPengNUS/rsome#23] [Q&A] [answered]
Hi, Thank you for the very useful library. 
I want to ask you how to include indicator constraints using RSOME. According to Gurobi support documents, there is a way of solving optimization problems with indicator constraints using Gurobi (https://support.gurobi.com/hc/en-us/articles/4414392016529-How-do-I-model-conditional-statements-in-Gurobi-). But I'm struggling to understand how to code it when using RSOME.

Best regards
Buddi

**Comment by XiongPengNUS** [comment, marked as answer]
RSOME does not provide direct support for indicator constraints, but you can follow the inequalities given at the website you provided to construct the constraints:

<img width="211" alt="image" src="https://github.com/XiongPengNUS/rsome/assets/41470721/a8cd9e83-4512-4f50-9b65-64bcd1ce28ff">

where $M$ can be defined to be a sufficiently large constant.
